src/utils.cpp

```diff
@@ -308,6 +308,14 @@ std::vector<std::string> split_string(const std::string &str,
   return elems;
 }
 
+std::string realpath(std::string const& path) {
+  int exist = ::access(path.c_str(), F_OK);
+  if (exist != 0) {
```

Comment:
nit: [access](https://man7.org/linux/man-pages/man2/access.2.html) returns -1 on error so let's check it directly:
```suggestion
  if (exist == -1) {
```

Reply:
Fix it , thanks.